instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -103,19 +103,14 @@ def detect_unmatched_pairs(obfuscated)
           end
 
           def database_span_name(sql)
-            # Setting span name to the SQL query without obfuscation would
-            # result in PII + cardinality issues.
-            # First attempt to infer the statement type then fallback to
-            # current Otel approach {database.component_name}.{database_instance_name}
-            # https://github.com/open-telemetry/opentelemetry-python/blob/39fa078312e6f41c403aa8cad1868264011f7546/ext/opentelemetry-ext-dbapi/tests/test_dbapi_integration.py#L53
-            # This creates span names like mysql.default, mysql.replica, postgresql.staging etc.
-
-            statement_type = extract_statement_type(sql)
-
-            return statement_type unless statement_type.nil?
-
-            # fallback
-            'mysql'
+            # TODO: When https://github.com/github/trilogy/pull/48/commits/6e2377ae2d737962a7a1a46b8f63438868034c7b is released,
+            # we will have access to the database name, so we can support :span_name configs of :db_name and :db_operation_and_name.
+            case config[:span_name]
+            when :statement_type
+              extract_statement_type(sql)
+            when :db_operation
+              OpenTelemetry::Instrumentation::Trilogy.attributes['db.operation']
```

Comment:
Does this mean it is always user supplied?

Reply:
IDK. I am not familiar with the motivation to add it to the mysql2 instrumentation so I was hoping to learn more about why we want to add a config option for this.

Reply:
From the [PR](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/222) where this option was added to Mysql2:
> The db_operation_and_name option additionally looks for db.operation in the OpenTelemetry::Instrumentation::Mysql2.attributes (from the current context). This allows callers to add db.operation to the context, if they know it.

[This comment](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/222#discussion_r1053552076) also points to the OTel spec, which suggests that
>  If db.sql.table is not available due to its semantics, the span SHOULD be named <db.operation> <db.name>

So my understanding was that OTel is in favour of including the db operation in the span name, if it's made available.